buildingspy/development/validator.py

```diff
@@ -177,3 +179,377 @@ def _validateHTML(self, moFile):
             with open(htmlName, mode="w", encoding="utf-8") as f:
                 f.write(document)
         return (document, errors)
+    
+    def _recursive_glob(self, rootdir='.', suffix=''):
+        """ 
+        Return all files with given extension.
+    
+        :param rootdir: Root directory.
+        :param suffix: File extension.
+        :return: List of files with given extension.
+
+    
+         """
+        
+        return [os.path.join(rootdir, filename) for rootdir, dirnames,
+                filenames in os.walk(rootdir) for filename in filenames 
+                if (filename.endswith(suffix) 
+                    and ("ConvertBuildings_from" not in filename)) ]    
+    
+    def _check_experiment(self, name, val, value, model_path, mos_file):
+        """ 
+        Check experiment annotation parameters in mo file.
+    
+        :param name: Parameter name.
+        :param val: Value found in mo file.
+        :param value: Value found in mos file.
+        :param model_path: Path to mo file.
+        :param mos_file: Path to mos file.
+    
+         """
+        
+        if("*" in str(val)):
+            s = ("Found mo file=" + str(model_path) + " with experiment annotation " + self._capitalize_first(name) + ".\n" 
+                + self._capitalize_first(name) + " contains invalid expressions such as x * y. Only literal expressions are allowed " 
+                + "by JModelica and OpenModelica unit tests.\n")
+            raise ValueError(s) 
+    
+        delta = abs(eval(val) - eval(value))
+        
+        if (delta!=0):
+            s = ("Found mo file=" + str(model_path) + " with experiment annotation " + self._capitalize_first(name) + ".\n" 
+                + "The value of " + self._capitalize_first(name) + "="+str(val) +" is different from the (default) value="
+                + str(value)+" found in the mos file=" + str(mos_file) + ".\n")
+            raise ValueError(s)            
+            
+
+            
+    def _missing_parameter(self, name, value, model_path, mos_file):
+        """ 
+        Check missing experiment annotation parameter in mo file.
+    
+        :param name: Parameter name.
+        :param value: Value found in mos file.
+        :param model_path: Path to mo file.
+        :param mos_file: Path to mos file.
+    
+         """
+        
+        s = ("Found mo file=" + str(model_path) + " without parameter " 
+             + self._capitalize_first(name) + " defined.\n" 
+            + "The parameter name " + name + " is defined in the mos file=" 
+            + str(mos_file) + " and hence must be defined in the mo file.\n")
+        raise ValueError(s)
```

Comment:
Why do you need `str()` for some but not for other arguments? Maybe using `...format(...)` would be easier (and more standard than adding strings).